sw/eventListener.js

```diff
@@ -19,72 +19,117 @@ export function onPush (sw) {
     if (!payload) return
     const { tag } = payload.options
     event.waitUntil((async () => {
-      if (immediatelyShowNotification(payload)) {
+      if (immediatelyShowNotification(tag)) {
         setAppBadge(sw, ++activeCount)
+        // FIXME(iOS) this also doesn't work on iOS as expected. it displays multiple notifications with same tag.
+        //   due to missing proper tag support in Safari on iOS, we can't rely on the tag property to replace notifications.
+        //   see https://bugs.webkit.org/show_bug.cgi?id=258922 for more information
+        //   we therefore fetch all notifications with the same tag (+ manual filter),
+        //   close them and then we display the notification.
+        const notifications = await sw.registration.getNotifications({ tag })
+        notifications.filter(({ tag: nTag }) => nTag === tag).forEach(n => n.close())
         return await sw.registration.showNotification(payload.title, payload.options)
```

Comment:
Couldn't we just merge here too or no?

Reply:
By just doing this, we make the merging code less complicated. Since there is no data we need to merge for these tags: `TIP`, `FORWARDEDTIP`, `EARN`, `STREAK`.

This means these tags simply replace the previous notification.

The other notifications need to be aware how notifications of that tag are currently displayed to the user.